bugfix: allow to retrieve function-units from Table/Column

bwinkel opened the pull request:
This fixes issue #8424, but I'm not sure if it is the best (or even an acceptable) solution to the problem. 

EDIT: Fix #8424

mhvk commented on astropy/table/column.py:
```diff
@@ -697,7 +697,7 @@ def quantity(self):
         # if the column's values are non-numeric (like strings), while .view
         # will happily return a quantity with gibberish for numerical values
-        return Quantity(self, copy=False, dtype=self.dtype, order='A')
+        return Quantity(self, self.unit, copy=False, dtype=self.dtype, order='A', subok=True)
```
I was a bit surprised that you needed to pass `self.unit` here, and feel that is a bug in `Quantity`. But probably best addressed separately - see #8430.

Anyway, this is the right way to do it.

mhvk commented on astropy/table/tests/test_table.py:
```diff
@@ -144,6 +144,19 @@ def test_set_new_col_new_table_quantity(self, table_types):
         assert np.all(t['bb'] == 3)
         assert t['bb'].unit == u.m
 
+    def test_get_col_table_func_quantity(self, table_types):
```
This test is OK in principle, but given that there is no other test of `.quantity` in this file, I think it would be more logical to replace it with a test that works directly on the `Column` instance, by adding something to `test_columns` (just after `test_to_quantity`, or perhaps as part of that).

bwinkel commented:
Thanks a lot @mhvk! I've moved the test to ``test_column.py`` and added an entry to the changelog (so that you can review, if the text is OK). I'm not quite sure though, what you mean with "Also, could you add an entry until bug fixes for 2.0.12?". Should the changelog message appear only/additionally below "2.0.12 (unreleased)" in ``CHANGES.rst``? (I've never dealt with a bug-fix PR - my apologies for the inconvenience.)

mhvk commented:
@bwinkel - no worries. The text of the changelog entry looks fine, but it should indeed be under bug fixes for 2.0.12 (unreleased); the text will get propagated also to 3.1.2 as part of the release procedure.

Thanks!

bwinkel commented:
OK, I've moved the changelog entry.

Should I squash the commits into one (given that the amount of changes is rather small)?

mhvk commented:
p.s. Thanks!
